### PR TITLE
Add support for config objects

### PR DESCRIPTION
**Pull request by gewfy**
👋  Hey, thanks for a great package!

[We generate our Netlify CMS config object](https://github.com/29ki/29k/blob/main/cms/src/index.js#L17) and therefore can't use yaml. I don't know how common this is, as it's undocumented. But I thought it would be a good addition to your project as we really want to use this in our setup.

Let me know if you like it and I can update the `README.md` on this too.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -1,8 +1,11 @@
 import { generateTypes } from "./generate";
-import { loadConfig } from "./input";
+import { NetlifyCMSConfig, loadConfig } from "./input";
 import type { NetlifyTsOptions } from "./types";
 
-export const createNetlifyTypes = (input: string, options: NetlifyTsOptions = {}): string => {
+export const createNetlifyTypes = (
+  input: string | NetlifyCMSConfig,
```
Comment (jsaari97) [inline src/index.ts:6]: Please add `NetlifyCMSConfig` as an input option for the `createNetlifyTypesAsync` function as well.

File: src/input.ts [hunk 2]
```diff
@@ -3,21 +3,29 @@ import fs from "fs";
 import path from "path";
 import type { Collection } from "./types";
 
-interface YamlInput {
+export interface NetlifyCMSConfig {
```
Comment (jsaari97) [inline src/input.ts:6]: `NetlifyCMSConfig` type could be moved to the `types` directory, as it is referenced outside this file as well, either into its own file or appended to `types/fields.ts`.

File: src/input.ts [hunk 3]
```diff
@@ -3,21 +3,29 @@ import fs from "fs";
...
-    throw new Error("Invalid filetype, must be an yaml file");
+export const loadConfig = (config: string | NetlifyCMSConfig): Collection[] => {
+  let data;
+
+  if (typeof config === "object") {
+    data = config;
   }
 
-  const file = fs.readFileSync(path.resolve(process.cwd(), filePath), "utf8");
+  if (typeof config === "string") {
```
Comment (jsaari97) [inline src/input.ts:17]: The loading of a config file could be extracted into its own function, e.g. `loadConfigFile(filePath: string): NetlifyCMSConfig` for better readability.

**Review by jsaari97 — CHANGES_REQUESTED** [review]
Hello, thanks for the pull request!

This feature looks like a nice addition to the library. I added a few comments on suggested changes, once those are resolved and the documentation updated (nothing fancy, I'll have to refactor the documentation at some point anyhow), I'll happily merge it! :)

File: src/index.ts (hunk 1 above)
Comment (gewfy) [reply src/index.ts:6]: Doh, ofc!

**Comment by gewfy** [comment]
Nice to hear! :) 
Updated with the requested changes and wrote an example in the readme.

Let me know if you see any other obvious improvements.

**Review by jsaari97 — APPROVED** [review]

**Pull request merged**